app/code/Academy/SmsSubscription/etc/adminhtml/system.xml

```diff
@@ -19,6 +19,20 @@
                     <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                 </field>
             </group>
+            <group id="api" translate="label" type="text" sortOrder="10"
+                   showInDefault="1" showInWebsite="0" showInStore="0">
+                <label>API</label>
+                <field id="api_key" translate="label" type="text"
+                       sortOrder="1" showInDefault="1" showInWebsite="0" showInStore="0">
+                    <label>API KEY</label>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
```

Comment:
This is not necessary here - this applies only for Yes/No `select` fields like the `enabled` field, here the type is `text` so `<source_model>` is ignored. Same in the below field.